content/applications/inventory_and_mrp/barcode/setup/software.rst

```diff
@@ -1,101 +1,145 @@
 
-Odoo users are able to use GTIN barcodes to identify their products. However, since Odoo supports
-any numeric string as a barcode, it is also possible to define a custom barcode for internal use.
+.. tip::
+   To filter for the products that do not have barcodes yet, click the :icon:`fa-sort-desc`
+   :guilabel:`(Toggle Search Panel)` icon to add a Custom Filter where the :guilabel:`Barcode`
+   property is :guilabel:`is not set`.
 
-.. _barcode/setup/barcodelookup:
+   .. image:: software/barcode-filter-for-no-barcode.png
+      :alt: 'Add Custom Filter' pop-up with 'Barcode is not set' configured.
 
-Barcode Lookup
-==============
+.. _barcode/setup/location:
+
+From Barcode app
+----------------
+
+New products definitions can be added to the database product lists as well as having their on-hand
+inventory tracked from the **Barcode** app, if the :ref:`Barcode Lookup feature
+<barcode/setup/barcodelookup>` is enabled. To create a new receipt for scanning new products, choose
+one of two methods:
+
+#. From the **Barcode** front page, tap :guilabel:`Operations`, tap :guilabel:`Receipts`, and then
+   tap the :guilabel:`New` button.
+#. Scan a printed :guilabel:`Receipts (WHIN)` inventory command barcode.
+
+.. image:: software/barcode-new-receipt.png
+   :alt: Empty new receipt form.
 
-`Barcode Lookup <https://www.barcodelookup.com/>`_ allows you to scan (or enter) products' barcodes
-(:abbr:`UPC (Universal Product Code)`, :abbr:`EAN (European Article Number)`, or :abbr:`ISBN
-(International Standard Book Number)`) to automatically create them in your Odoo database, complete
-with product names, descriptions, images, categories, etc.
+Scanning a product that is not currently in inventory prompts a message that the product does not
+exist, with a button option to :guilabel:`Create New Product`. Pressing this button with search the
+`Barcode Lookup <https://www.barcodelookup.com/>`_ database for a product matching the code and
+format, create a new product definition in the Odoo database with the available information, and add
+that product to the receipt to track the quantity on hand in inventory.
 
-To use Barcode Lookup to scan and create products in Barcode, go to :menuselection:`Inventory -->
-Configuration --> Settings`, scroll down to the :guilabel:`Barcode` section, and enable
-:guilabel:`Stock Barcode Database`.
+.. image:: software/barcode-scan-for-new-product.png
+   :alt: New product confirmation dialog.
 
 .. note::
-   If your database is hosted on **Odoo.sh** or **on-premise**, you must also :ref:`configure
-   an API key <barcodelookup/configuration>`.
-   If your database is hosted on **Odoo Online**, you can use Barcode Lookup without further
-   configuration.
+   Even if a transfer order with a product created through **Barcode** is cancelled, the product
+   remains available in the products list in the **Inventory** app unless it is deleted.
+
+Print location barcodes
```

Comment:
This is a good start. 
Printing, in my mind, would come *after* *Location* configuration for products based on their individual form values, product categories, lots, etc. That information would be good to include here, using the barcode scanner to help, i.e., receive new products not previously configured in Odoo --> point the scanner at the Lot of products --> batch scan --> programs their location destination in Odoo as well as their inventory counts (something like this).